### ISSUE TITLE
Support TypeScript NodeNext module resolution

### ISSUE THREAD

**Issue by benasher44** [bpmn-io/min-dash#28]
### Is your feature request related to a problem? Please describe.

If you're using Node + ESM + TypeScript, packages that export an ESM build to correct ESM import syntax.

### Describe the solution you'd like

Update all imports and exports to have .js extensions.

### Describe alternatives you've considered

I'm not sure there are alternatives.

### Additional context

If you import this package in a project that is using this combination, you get errors like this:

```
node_modules/min-dash/dist/index.d.ts:2:15 - error TS2834: Relative import paths need explicit file extensions in ECMAScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Consider adding an extension to the import path.

export * from './collection';
              ~~~~~~~~~~~~~~
```

Also see https://arethetypeswrong.github.io/?p=min-dash%404.1.1

**Comment by nikku** [comment]
Happy to take a contribution to improve our types. Not sure what you refer to with 

> Update all imports and exports to have .js extensions.

We export dedicated types and bundled distribution. So this would not do a thing for you (I think).

**Comment by nikku** [comment]
The website you linked hints that explicit `exports` declaration is required instead of resorting to `main` and `module` fields used in this library.

**Comment by benasher44** [comment]
Ah yeah sorry I guess there are two issues, and I likely can only see one right now. The first is what tsc is telling me: missing extensions. Once that's fixed, we'll likely run into the issues called out by arethetypeswrong.

I'll see if I can start with contributing .js extensions

**Comment by nikku** [comment]
[This](https://stackoverflow.com/questions/72457791/typescript-packages-that-ship-with-mjs-and-d-ts-but-without-d-mts-how-to-i) may be additional helpful context.

We need to decide what our "type target" is, or we need to define dedicated type definitions for CJS _and_ ESM.

**Comment by benasher44** [comment]
Yep exactly. One strategy would be to copy all of the .d.ts files as .d.cts. Since package.json's type is set to "module", this should result in ESM packages reading .d.ts files and CommonJS packages reading .d.cts files. The PR does not fix it for CJS + node16 resolution, but it will improve the situation overall.

**Comment by nikku** [comment]
This should be fully addressed with [`v4.2.0`](https://github.com/bpmn-io/min-dash/releases/tag/v4.2.0).

**Comment by benasher44** [comment]
Amazing, thank you!
